src/common/bots/Bot.cpp

```diff
@@ -0,0 +1,73 @@
+#include "Bot.hpp"
+
+Bot::Bot(){};
+
+/* This strategy will move towards the closest mine which has a resource the bot needs to win.
+ * Once it has enough of a resource, it will move to the next closest mine with a resource it needs.
+ */
+std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
+Bot::getNearestGreedyMove(std::vector<std::shared_ptr<Mine>> mines,
+                          std::array<uint32_t, RESOURCE_TYPE_COUNT> current_resources,
+                          sf::Vector2f bot_center,
+                          std::unordered_map<uint32_t, uint32_t> mine_id_to_resource_count) {
+    std::sort(mines.begin(), mines.end(),
+              [bot_center](std::shared_ptr<Mine> mine_a, std::shared_ptr<Mine> mine_b) {
+                  auto dist_a = VectorUtil::distance(bot_center, mine_a->getCenter());
+                  auto dist_b = VectorUtil::distance(bot_center, mine_b->getCenter());
+                  return dist_a < dist_b;
+              });
+
+    for (auto mine : mines) {
```

Comment:
rather than copying the shared_ptr here (costs cpu because you need to +1 the ptr count), I would take a reference in the loop by doing 
```
for(auto& mine : mines)
```